flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/UpsertKafkaDynamicTableFactory.java

```diff
@@ -155,6 +161,20 @@ public DynamicTableSink createDynamicTableSink(Context context) {
 
         Integer parallelism = tableOptions.get(FactoryUtil.SINK_PARALLELISM);
 
+        KafkaDynamicSink.SinkFunctionProviderCreator creator;
+        Duration interval = tableOptions.get(SINK_BUFFER_FLUSH_INTERVAL);
+        Integer batchSize = tableOptions.get(SINK_BUFFER_FLUSH_MAX_ROWS);
+        if (batchSize > 0 && interval.toMillis() > 0) {
```

Comment:
We should enable buffering when one of them is larger than zero.

Reply:
Solved. 
Add validation in TableFactory